client/settings/check-starcraft-path.js

```diff
@@ -42,20 +44,28 @@ async function checkHash(path, validHashes) {
 // be checked for other copies. If downgradePath contains files that match the correct hashes, this
 // will be counted as having the correct version, but `downgradePath` will be true.
 export async function checkStarcraftPath(dirPath, downgradePath) {
-  const requiredFiles = ['starcraft.exe', 'storm.dll', 'local.dll', 'stardat.mpq', 'broodat.mpq']
+  const requiredFiles = ['starcraft.exe', 'storm.dll', 'stardat.mpq', 'broodat.mpq']
 
   try {
     await Promise.all(requiredFiles.map(f => accessAsync(path.join(dirPath, f), fs.constants.R_OK)))
   } catch (err) {
     return { path: false, version: false, downgradePath: false }
   }
 
+  // Due to 1.19 version moving local.dll to a separate folder, we need to handle it separately
```

Comment:
Just do 2 checks:
- In the root folder (aka valid for 1.16)
- At locales/*/local.dll

Globbing everything under StarCraft's folder is potentially expensive, people have large map collections and such. This setup also avoids the need to check for the first case in the matches, which seems very error prone